File: platform/Leetcode/src/Insert.java
// Solution in Java
import java.util.*;

public class Insert {

    public static void main (String[] args){
        System.out.println("Enter a list of integers separated by commas: ");
        Scanner input1 = new Scanner(System.in); 
        String numberList = input1.nextLine();
        
        System.out.println("Now enter a target number to be inserted: ");
        Scanner input2 = new Scanner(System.in); 
        String targetNumber = input2.nextLine();
        
        int targetIndex = searchInsert(numberList, targetNumber);
        String position = Integer.toString(targetIndex);
        System.out.println("The target number should go on index " + position + " of list above");
    }
    
    public static int searchInsert (String numbers, String target) {    
        String[] items = numbers.split(",");
        List<String> itemList = new ArrayList<String>(Arrays.asList(items));

        for (int i=0; i<itemList.size(); i++ ) {
            if (Integer.valueOf(target) == Integer.valueOf(itemList.get(i)) ) {
                return i;
            } else {
                if (Integer.valueOf(target) > Integer.valueOf(itemList.get(i))) {
                    continue;
                } else {
                    return i;
                }
            }
        }
        int output = itemList.size();
        return output;    
    }
}

File: platform/Leetcode/src/Insert.py
# Solution 2
def search_insert(num_list, target):
    for ind, num in enumerate(num_list):
        if target> num:
            continue
        else:
            return ind
    return ind+1

def main():
    number_list = input("Enter a list of numbers separated by commas: ")
    target = input("Enter a target number: ")
    integer_list = [int(each.strip()) for each in number_list.split(",")]
    position = search_insert(integer_list, int(target))
    if position+1 == 2:
        ith = 'nd'
    elif position+1 == 3:
        ith = 'rd'
    else: 
        ith = 'th'
    print("The target should be inserted at the {}-{} position in the list".format(position+1, ith))
    
    
if __name__=='__main__':
   main()
